File: libs/jgdsl/src/gdsl/asm/operand/Immediate.java
package gdsl.asm.operand;

import gdsl.asm.Visitor;

public class Immediate extends Operand {
  private long value;
  
  public long getValue() {
    return value;
  }

  /**
   * @param value
   */
  public Immediate (long value) {
    super();
    this.value = value;
  }

  @Override public String toString () {
    return Long.toString(value);
  }

  @Override public void accept (Visitor v) {
    v.visit(this);
  }
}

File: libs/cppgdsl/src/assembly/operand/immediate.cpp
#include "cppgdsl/assembly/operand/immediate.h"
#include "cppgdsl/assembly/operand/operand_visitor.h"

namespace gdsl {
namespace assembly {

void immediate::put(std::ostream& out) const { out << imm; }

void immediate::accept(operand_visitor& ov) const { ov.visit(this); }

bool immediate::operator==(operand const& o) const {
  bool equals = false;
  operand_visitor v(true);
  v._([&](immediate const* o) { equals = this->imm == o->imm; });
  o.accept(v);
  return equals;
}

}  // namespace assembly
}  // namespace gdsl
